### ISSUE TITLE
Default value for column timestamp (postgres) defines each time on startup

### ISSUE THREAD

**Issue by saewave** [typeorm/typeorm#10328]
### Issue description

Column described as "timestamp without time zone" with default value "(now())::timestamp(0) without time zone" produces "ALTER TABLE ... ALTER COLUMN ... SET DEFAULT" query

### Expected Behavior

Should set this value only once.

### Actual Behavior

Generates  `query: ALTER TABLE "job" ALTER COLUMN "addDate" SET DEFAULT (now())::timestamp(0) without time zone` each time when app runs

### Steps to reproduce

Create entity with column like this:

```
@Entity()
export class Job {
    @PrimaryGeneratedColumn('uuid')
    id: string;

    @Column({ type: 'timestamp without time zone', default: () => '(now())::timestamp(0) without time zone' })
    addDate: boolean;
....
}
```
This will generate table `Job` with column `addDate`  
Execute sql to retrieve table columns with attributes:
```
SELECT COLUMNS.*,
	PG_CATALOG.COL_DESCRIPTION(('"' || TABLE_CATALOG || '"."' || TABLE_SCHEMA || '"."' || TABLE_NAME || '"')::REGCLASS::OID,

													ORDINAL_POSITION) AS DESCRIPTION,
	('"' || "udt_schema" || '"."' || "udt_name" || '"')::"regtype" AS "regtype",
	PG_CATALOG.FORMAT_TYPE("col_attr"."atttypid",

													"col_attr"."atttypmod") AS "format_type"
FROM "information_schema"."columns"
LEFT JOIN "pg_catalog"."pg_attribute" AS "col_attr" ON "col_attr"."attname" = "columns"."column_name"
AND "col_attr"."attrelid" =
	(SELECT "cls"."oid"
		FROM "pg_catalog"."pg_class" AS "cls"
		LEFT JOIN "pg_catalog"."pg_namespace" AS "ns" ON "ns"."oid" = "cls"."relnamespace"
		WHERE "cls"."relname" = "columns"."table_name"
			AND "ns"."nspname" = "columns"."table_schema" )
WHERE ("table_schema" = 'public'
							AND "table_name" = 'job')
```
Result:

[data-1693659215550.csv](https://github.com/typeorm/typeorm/files/12504007/data-1693659215550.csv)

Run app with debug mode and you will see something like this:
```
[12:02:12 PM] Starting compilation in watch mode...

[12:02:27 PM] Found 0 errors. Watching for file changes.

[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [NestFactory] Starting Nest application...
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] AppModule dependencies initialized +41ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] AppDBModule dependencies initialized +1ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] DiscoveryModule dependencies initialized +1ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] ConfigHostModule dependencies initialized +1ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] ScheduleModule dependencies initialized +2ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] ConfigModule dependencies initialized +1ms
[Nest] 9418  - 09/02/2023, 12:02:38 PM     LOG [InstanceLoader] JwtModule dependencies initialized +174ms
query: SELECT * FROM current_schema()
query: CREATE EXTENSION IF NOT EXISTS "uuid-ossp"
query: SELECT version();
query: START TRANSACTION
query: SELECT * FROM current_schema()
query: SELECT * FROM current_database()
query: SELECT "table_schema", "table_name" FROM "information_schema"."tables" WHERE ("table_schema" = 'public' AND "table_name" = 'user') OR ("table_schema" = 'public' AND "table_name" = 'orpc') OR ("table_schema" = 'public' AND "table_name" = 'network') OR ("table_schema" = 'public' AND "table_name" = 'contract') OR ("table_schema" = 'public' AND "table_name" = 'account') OR ("table_schema" = 'public' AND "table_name" = 'contract_function') OR ("table_schema" = 'public' AND "table_name" = 'job') OR ("table_schema" = 'public' AND "table_name" = 'scheduler_job') OR ("table_schema" = 'public' AND "table_name" = 'scheduler') OR ("table_schema" = 'public' AND "table_name" = 'action_log') OR ("table_schema" = 'public' AND "table_name" = 'gas_statistic')
query: SELECT TRUE FROM information_schema.columns WHERE table_name = 'pg_class' and column_name = 'relispartition'
query: SELECT columns.*, pg_catalog.col_description(('"' || table_catalog || '"."' || table_schema || '"."' || table_name || '"')::regclass::oid, ordinal_position) AS description, ('"' || "udt_schema" || '"."' || "udt_name" || '"')::"regtype" AS "regtype", pg_catalog.format_type("col_attr"."atttypid", "col_attr"."atttypmod") AS "format_type" FROM "information_schema"."columns" LEFT JOIN "pg_catalog"."pg_attribute" AS "col_attr" ON "col_attr"."attname" = "columns"."column_name" AND "col_attr"."attrelid" = ( SELECT "cls"."oid" FROM "pg_catalog"."pg_class" AS "cls" LEFT JOIN "pg_catalog"."pg_namespace" AS "ns" ON "ns"."oid" = "cls"."relnamespace" WHERE "cls"."relname" = "columns"."table_name" AND "ns"."nspname" = "columns"."table_schema" ) WHERE ("table_schema" = 'public' AND "table_name" = 'job') OR ("table_schema" = 'public' AND "table_name" = 'scheduler') OR ("table_schema" = 'public' AND "table_name" = 'user') OR ("table_schema" = 'public' AND "table_name" = 'orpc') OR ("table_schema" = 'public' AND "table_name" = 'gas_statistic') OR ("table_schema" = 'public' AND "table_name" = 'network') OR ("table_schema" = 'public' AND "table_name" = 'account') OR ("table_schema" = 'public' AND "table_name" = 'contract_function') OR ("table_schema" = 'public' AND "table_name" = 'scheduler_job') OR ("table_schema" = 'public' AND "table_name" = 'contract') OR ("table_schema" = 'public' AND "table_name" = 'action_log')
query: SELECT "ns"."nspname" AS "table_schema", "t"."relname" AS "table_name", "cnst"."conname" AS "constraint_name", pg_get_constraintdef("cnst"."oid") AS "expression", CASE "cnst"."contype" WHEN 'p' THEN 'PRIMARY' WHEN 'u' THEN 'UNIQUE' WHEN 'c' THEN 'CHECK' WHEN 'x' THEN 'EXCLUDE' END AS "constraint_type", "a"."attname" AS "column_name" FROM "pg_constraint" "cnst" INNER JOIN "pg_class" "t" ON "t"."oid" = "cnst"."conrelid" INNER JOIN "pg_namespace" "ns" ON "ns"."oid" = "cnst"."connamespace" LEFT JOIN "pg_attribute" "a" ON "a"."attrelid" = "cnst"."conrelid" AND "a"."attnum" = ANY ("cnst"."conkey") WHERE "t"."relkind" IN ('r', 'p') AND (("ns"."nspname" = 'public' AND "t"."relname" = 'job') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'scheduler') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'user') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'orpc') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'gas_statistic') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'network') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'account') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'contract_function') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'scheduler_job') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'contract') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'action_log'))
query: SELECT "ns"."nspname" AS "table_schema", "t"."relname" AS "table_name", "i"."relname" AS "constraint_name", "a"."attname" AS "column_name", CASE "ix"."indisunique" WHEN 't' THEN 'TRUE' ELSE'FALSE' END AS "is_unique", pg_get_expr("ix"."indpred", "ix"."indrelid") AS "condition", "types"."typname" AS "type_name" FROM "pg_class" "t" INNER JOIN "pg_index" "ix" ON "ix"."indrelid" = "t"."oid" INNER JOIN "pg_attribute" "a" ON "a"."attrelid" = "t"."oid"  AND "a"."attnum" = ANY ("ix"."indkey") INNER JOIN "pg_namespace" "ns" ON "ns"."oid" = "t"."relnamespace" INNER JOIN "pg_class" "i" ON "i"."oid" = "ix"."indexrelid" INNER JOIN "pg_type" "types" ON "types"."oid" = "a"."atttypid" LEFT JOIN "pg_constraint" "cnst" ON "cnst"."conname" = "i"."relname" WHERE "t"."relkind" IN ('r', 'p') AND "cnst"."contype" IS NULL AND (("ns"."nspname" = 'public' AND "t"."relname" = 'job') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'scheduler') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'user') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'orpc') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'gas_statistic') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'network') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'account') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'contract_function') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'scheduler_job') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'contract') OR ("ns"."nspname" = 'public' AND "t"."relname" = 'action_log'))
query: SELECT "con"."conname" AS "constraint_name", "con"."nspname" AS "table_schema", "con"."relname" AS "table_name", "att2"."attname" AS "column_name", "ns"."nspname" AS "referenced_table_schema", "cl"."relname" AS "referenced_table_name", "att"."attname" AS "referenced_column_name", "con"."confdeltype" AS "on_delete", "con"."confupdtype" AS "on_update", "con"."condeferrable" AS "deferrable", "con"."condeferred" AS "deferred" FROM ( SELECT UNNEST ("con1"."conkey") AS "parent", UNNEST ("con1"."confkey") AS "child", "con1"."confrelid", "con1"."conrelid", "con1"."conname", "con1"."contype", "ns"."nspname", "cl"."relname", "con1"."condeferrable", CASE WHEN "con1"."condeferred" THEN 'INITIALLY DEFERRED' ELSE 'INITIALLY IMMEDIATE' END as condeferred, CASE "con1"."confdeltype" WHEN 'a' THEN 'NO ACTION' WHEN 'r' THEN 'RESTRICT' WHEN 'c' THEN 'CASCADE' WHEN 'n' THEN 'SET NULL' WHEN 'd' THEN 'SET DEFAULT' END as "confdeltype", CASE "con1"."confupdtype" WHEN 'a' THEN 'NO ACTION' WHEN 'r' THEN 'RESTRICT' WHEN 'c' THEN 'CASCADE' WHEN 'n' THEN 'SET NULL' WHEN 'd' THEN 'SET DEFAULT' END as "confupdtype" FROM "pg_class" "cl" INNER JOIN "pg_namespace" "ns" ON "cl"."relnamespace" = "ns"."oid" INNER JOIN "pg_constraint" "con1" ON "con1"."conrelid" = "cl"."oid" WHERE "con1"."contype" = 'f' AND (("ns"."nspname" = 'public' AND "cl"."relname" = 'job') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'scheduler') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'user') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'orpc') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'gas_statistic') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'network') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'account') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'contract_function') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'scheduler_job') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'contract') OR ("ns"."nspname" = 'public' AND "cl"."relname" = 'action_log')) ) "con" INNER JOIN "pg_attribute" "att" ON "att"."attrelid" = "con"."confrelid" AND "att"."attnum" = "con"."child" INNER JOIN "pg_class" "cl" ON "cl"."oid" = "con"."confrelid"  AND "cl"."relispartition" = 'f'INNER JOIN "pg_namespace" "ns" ON "cl"."relnamespace" = "ns"."oid" INNER JOIN "pg_attribute" "att2" ON "att2"."attrelid" = "con"."conrelid" AND "att2"."attnum" = "con"."parent"
query: SELECT "udt_schema", "udt_name" FROM "information_schema"."columns" WHERE "table_schema" = 'public' AND "table_name" = 'scheduler' AND "column_name"='daysOfWeek'
query: SELECT "e"."enumlabel" AS "value" FROM "pg_enum" "e" INNER JOIN "pg_type" "t" ON "t"."oid" = "e"."enumtypid" INNER JOIN "pg_namespace" "n" ON "n"."oid" = "t"."typnamespace" WHERE "n"."nspname" = 'public' AND "t"."typname" = 'int2'
query: SELECT * FROM "information_schema"."tables" WHERE "table_schema" = 'public' AND "table_name" = 'typeorm_metadata'
query: ALTER TABLE "contract" ALTER COLUMN "addDate" SET DEFAULT (now())::timestamp(0) without time zone
query: ALTER TABLE "account" ALTER COLUMN "addDate" SET DEFAULT now()::timestamp(0)
query: ALTER TABLE "job" ALTER COLUMN "addDate" SET DEFAULT (now())::timestamp(0) without time zone
query: ALTER TABLE "scheduler" ALTER COLUMN "addDate" SET DEFAULT (now())::timestamp(0) without time zone
query: ALTER TABLE "action_log" ALTER COLUMN "addDate" SET DEFAULT now()::timestamp(0)
query: ALTER TABLE "gas_statistic" ALTER COLUMN "addDate" SET DEFAULT now()::timestamp(0)
query: COMMIT
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmCoreModule dependencies initialized +935ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +1ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +1ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] TypeOrmModule dependencies initialized +0ms
[Nest] 9418  - 09/02/2023, 12:02:39 PM     LOG [InstanceLoader] NetworkModule dependencies initialized +0ms
........
[Nest] 9418  - 09/02/2023, 12:02:40 PM     LOG [NestApplication] Nest application successfully started +20ms

```

Queries like this:
query: ALTER TABLE "job" ALTER COLUMN "addDate" SET DEFAULT (now())::timestamp(0) without time zone
Executes each time when apps starts

### My Environment

| Dependency          | Version  |
"dependencies": {
    "@nestjs/common": "^10.0.0",
    "@nestjs/config": "^3.0.0",
    "@nestjs/core": "^10.0.0",
    "@nestjs/jwt": "^10.1.0",
    "@nestjs/platform-express": "^10.0.0",
    "@nestjs/schedule": "^3.0.2",
    "@nestjs/swagger": "7.1.7",
    "@nestjs/typeorm": "^10.0.0",
    "bcrypt": "^5.1.0",
    "class-transformer": "^0.5.1",
    "class-validator": "^0.14.0",
    "cron": "^2.4.1",
    "ethers": "^5.7.2",
    "nodemon": "^3.0.1",
    "pg": "^8.11.2",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^7.8.1",
    "typeorm": "^0.3.17",
    "typeorm-naming-strategies": "^4.1.0",
    "zksync-web3": "^0.14.3"
  },
  "devDependencies": {
    "@nestjs/cli": "^10.0.0",
    "@nestjs/schematics": "^10.0.0",
    "@nestjs/testing": "^10.0.0",
    "@types/bcrypt": "^5.0.0",
    "@types/cron": "^2.4.0",
    "@types/express": "^4.17.17",
    "@types/jest": "^29.5.2",
    "@types/node": "^20.4.9",
    "@types/supertest": "^2.0.12",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "eslint": "^8.42.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-plugin-prettier": "^5.0.0",
    "jest": "^29.5.0",
    "prettier": "^3.0.0",
    "source-map-support": "^0.5.21",
    "supertest": "^6.3.3",
    "ts-jest": "^29.1.0",
    "ts-loader": "^9.4.3",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.2.0",
    "typescript": "^5.1.3"
  },
| Operating System    |          |
| Node.js version     | v18.13.0.  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.
